pandas/io/formats/style.py

```diff
@@ -379,6 +383,309 @@ def to_excel(
+
+        >>> s = df.style.highlight_max(axis=None,
+        ...                            props='cellcolor:{red}; bfseries: ;')
+        >>> s.to_latex()
+
+        Internally these structured LaTeX ``(<command>, <options>)`` pairs
+        are translated to the
+        ``display_value`` with the default structure:
+        ``\<command><options> <display_value>``.
+        Where there are multiple commands the latter is nested recursively, so that
+        the above example highlighed cell is rendered as
+        ``\cellcolor{red} \bfseries 4``.
+
+        Occasionally this format does not suit the applied command, or
+        combination of LaTeX packages that is in use, so additional flags can be
+        added to the ``<options>``, within the tuple, to result in different
+        positions of required braces (the **default** being the same as ``--nowrap``):
+
+        =================================== ============================================
+        Tuple Format                           Output Structure
+        =================================== ============================================
+        (<command>,<options>)               \\<command><options> <display_value>
+        (<command>,<options> ``--nowrap``)  \\<command><options> <display_value>
+        (<command>,<options> ``--rwrap``)   \\<command><options>{<display_value>}
+        (<command>,<options> ``--wrap``)    {\\<command><options> <display_value>}
+        (<command>,<options> ``--lwrap``)   {\\<command><options>} <display_value>
+        (<command>,<options> ``--dwrap``)   {\\<command><options>}{<display_value>}
+        =================================== ============================================
+
+        For example the `textbf` command for font-weight
+        should always be used with `--rwrap` so ``('textbf', '--rwrap')`` will render a
+        working cell, wrapped with braces, as ``\textbf{<display_value>}``.
+
+        A more comprehensive example is as follows:
+
+        >>> df = pd.DataFrame([[1, 2.2, "dogs"], [3, 4.4, "cats"], [2, 6.6, "cows"]],
+        ...                   index=["ix1", "ix2", "ix3"],
+        ...                   columns=["Integers", "Floats", "Strings"])
+        >>> s = df.style.highlight_max(
+        ...     props='cellcolor:[HTML]{FFFF00}; color:{red};'
+        ...           'textit:--rwrap; textbf:--rwrap;'
+        ... )
+        >>> s.to_latex()
+
+        .. figure:: ../../_static/style/latex_1.png
+
+        **Table Styles**
+
+        Internally Styler uses its ``table_styles`` object to parse the
+        ``column_format``, ``position``, ``position_float``, ``hrules`` and ``label``
+        input arguments. There is additional scope to add custom LaTeX commands,
+        which are included and positioned
+        immediately above the '\\begin{tabular}' command. For example to add odd and
+        even row coloring, from the {colortbl} package, use:
```

Comment:
It took me a bit to understand this section, but made sense after playing with some examples. My takeaways were:

 - In general, set_table_styles can be used to insert latex commands between `\begin{table}` and `\begin{tabular}`
 - Special exception is made for selector being `toprule`, `midrule`, or `bottomrule`, which can be used to insert commands in the corresponding section of `tabular` environment.
 - Users need to know that internally `table_styles` is used for the various arguments (e.g. ``column_format``) because the user's usage of ``table_styles`` could be in conflict with the arguments usage.

The first bullet is already mentioned currently, but I think could be made a little more explicit by saying something like:

 > '\\begin{tabular}' command that can be accessed by using ``set_table_styles``.

For the 2nd bullet, is my list exhaustive? If so, it wasn't clear to me from the documentation. Similarly with the 3rd bullet, if my statement is correct (and is the motivation for mentioning this line), I think it would be good to state something like it explicitly.

Reply:
the list is:

 - 'position': has a specific placement: \begin{table}[**position**]
 - 'position_float': has a specific placement: **after** \begin{table}[position]
 - 'caption': is the first of the commands between {table} and {tabular} (becuase 'label' **must** follow it)
 - 'label' and 'udf_commands': commands between {table} and {tabular} order based on user coding.
 - 'column_format': has specific placement: \begin{tabular}{**column_format**}
 - 'toprule', 'midrule', 'bottomrule': specific placement in the {tabular} environment.

And that's your lot.

I will have a go at a re-write.

Reply:
While it is common to put the table caption above the table, there are also styles that require the caption to be placed at the bottom. It would be great to have this configurable, but I also don't want to further increase the scope of the pull request. This might be something for a follow-up.

Reply:
once this is approved add it as a github `issue`, and it will be easy to deal with afterwards, probably a new keyword `caption_position: {'top', 'bottom'}` or similar.

Reply:
Thanks @attach68 - I think this section looks great. Going to take one more look through in the next day.